"$topN"' does not exist in type 'AccumulatorOperator'

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

6.6.0

### Node.js version

16.13.0

### MongoDB server version

6.0.1

### Description

I am getting the same issue as reported in : [12026](https://github.com/Automattic/mongoose/discussions/12026)

Error
```
src/models/price_event.ts:57:21 - error TS2322: Type '{ $topN: { output: string[]; sortBy: { txnDate: number; }; n: number; }; }' is not assignable to type 'AccumulatorOperator'.
  Object literal may only specify known properties, and '"$topN"' does not exist in type 'AccumulatorOperator'
```

Query:
   ```ts
 const query:PipelineStage[] = [
        {
            "$group" : {
                "_id" : {
                    "site" : "$metaData.site",
                    "product" : "$metaData.product"
                },
                "lastPrices" : {
                    "$topN" : {
                        "output" : [
                            "$txnDate",
                            "$cpl"
                        ],
                        "sortBy" : {
                            "txnDate" : -1.0
                        },
                        "n" : 7.0
                    }
                }
            }
        }, 
        {
            "$group" : {
                "_id" : {
                    "site" : "$_id.site"
                },
                "results" : {
                    "$push" : {
                        "product" : "$_id.product",
                        "priceTime" : "$lastPrices"
                    }
                }
            }
        }, 
        {
            "$project" : {
                "site" : "$_id.site",
                "productPriceTime" : "$results",
                "_id" : 0.0
            }
        }, 
        {
            "$sort" : {
                "site" : 1.0
            }
        }
    ];
```

However if I rewrite the query, the error is resolved

```ts

    const e:Expression = {
        "$topN" : {
            "output" : [
                "$txnDate",
                "$cpl"
            ],
            "sortBy" : {
                "txnDate" : -1.0
            },
            "n" : 7.0
        }        
    }

    const query:PipelineStage[] = 
    [
        {
            "$group" : {
                "_id" : {
                    "site" : "$metaData.site",
                    "product" : "$metaData.product"
                },
                "lastPrices" : e
            }
        }, 
        {
            "$group" : {
                "_id" : {
                    "site" : "$_id.site"
                },
                "results" : {
                    "$push" : {
                        "product" : "$_id.product",
                        "priceTime" : "$lastPrices"
                    }
                }
            }
        }, 
        {
            "$project" : {
                "site" : "$_id.site",
                "productPriceTime" : "$results",
                "_id" : 0.0
            }
        }, 
        {
            "$sort" : {
                "site" : 1.0
            }
        }
    ];
```

### Steps to Reproduce

tsc 

### Expected Behavior

Expect $topN to be supported in main query body, not as separate Expression.

--- a/test/types/expressions.test.ts
+++ b/test/types/expressions.test.ts
@@ -1,3 +1,4 @@
+import { PipelineStage } from 'mongoose';
 import type { Expression } from 'mongoose';
 import { expectError, expectType } from 'tsd';
 
@@ -220,4 +221,22 @@ const switchExpr: Expression.Switch = {
 
 (function gh12149() {
   const count: Expression.Count = { $count: '$value' };
+})();
+
+(function gh12417() {
+  const query: PipelineStage[] = [
+    {
+      $group: {
+        group: {
+          $topN: {
+            output: ['$field'],
+            sortBy: {
+              field: -1.0
+            },
+            n: 7.0
+          }
+        }
+      }
+    }
+  ];
 })();

--- a/types/expressions.d.ts
+++ b/types/expressions.d.ts
@@ -2835,7 +2835,8 @@ declare module 'mongoose' {
     Expression.Push |
     Expression.StdDevPop |
     Expression.StdDevSamp |
-    Expression.Sum;
+    Expression.Sum |
+    Expression.TopN;
 
   export type tzExpression = UTCOffset | StringExpressionOperatorReturningBoolean | string;
 

